File: 900/CF2225B_Alternating_String.java

import java.util.*;

public class CF2225B_Alternating_String {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int t = sc.nextInt();

        while (t-- > 0) {
            String s = sc.next();

            int count = 0;

            for (int i = 0; i < s.length() - 1; i++) {
                if (s.charAt(i) == s.charAt(i + 1)) {
                    count++;
                }
            }

            if (count <= 2) {
                System.out.println("YES");
            } else {
                System.out.println("NO");
            }
        }

        sc.close();
    }
}

File: 900/CF2225B_Alternating_String.py
t = int(input())

for _ in range(t):
    s = input().strip()

    count = 0

    for i in range(len(s) - 1):
        if s[i] == s[i + 1]:
            count += 1

    if count <= 2:
        print("YES")
    else:
        print("NO")